walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,55 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> requiredAttendees = request.getAttendees();
+    long meetingDuration = request.getDuration();
+    
+    List<TimeRange> eventsSortedByStartTime = new ArrayList<>();
+    List<TimeRange> eventsSortedByEndTime = new ArrayList<>();
+
+    for (Event event : events) {
+      // Add event to events lists if meeting request contains attendees of the event  
+      if (!Collections.disjoint(requiredAttendees, event.getAttendees())) {
+        eventsSortedByStartTime.add(event.getWhen());
+        eventsSortedByEndTime.add(event.getWhen());
+      }
+    }
+    Collections.sort(eventsSortedByStartTime, TimeRange.ORDER_BY_START);
+    Collections.sort(eventsSortedByEndTime, TimeRange.ORDER_BY_END);
+
+    List<TimeRange> meetingQueryOptions = new ArrayList<>();
+
+    if (meetingDuration > TimeRange.WHOLE_DAY.duration())
+      return meetingQueryOptions;
```

Comment:
This check could happen at the beginning of the method to avoid the list allocations and sorting if the meeting is longer than a day. Also, what if the duration is 0?